### PR TITLE
GF Signup: Experiment social signups with the onboarding-pm flow

### PR DESCRIPTION
**Pull request by ddc22**
Fixes Automattic/growth-foundations#271
Experiment: 21504-explat-experiment
## Proposed Changes

* Loads/assigns the experiment on load of the route `/start/onboarding-pm`
* Checks for variant and redirects to the brand new onboarding flow `/start/onboarding-pm-social`

## Testing Instructions

Case 1
* Assign yourself to treatment
* Go to `/start/onboarding-pm` 
* You should see the Social First passwordless signup page

Case 2
* Assign yourself to treatment
* Go to `/start/onboarding-pm` 
* You should see the legacy signup page

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by ddc22** [comment]
@aaronyan @southp curious to know your opinion about my approach to implement this experiment :)

**Review by aaronyan — COMMENTED** [review]
What happens if someone in `control` visits `onboarding-pm-social` and vice versa? Are they redirected to the appropriate page due to their assignment?

**Comment by ddc22** [comment]
> What happens if someone in control visits onboarding-pm-social and vice versa?

Ah good point! I can make that change 👍

**Comment by aaronyan** [comment]
Thanks Dehan! I prefer if the experiment occurred on the same page and the contents change based on assignment. This would prevent users from manually seeing they are in an experiment. Would that be possible?

**Comment by ddc22** [comment]
@aaronyan this might take a bit longer since the styling of the component is tied mostly to the signup step user-social. I will see what I can do. (This was my initial attempt actually)

Some feedback here @escapemanuele to try to further encapsulate styles inside UI components instead of injecting SASS based on global page based selectors. ( I am however not aware the context involved here so apologies if I am suggesting something impractical ).


Simply changing the `isSocialFirst` inside the  `user step` does not intuitively show the proper social user page.
image

---

Update: This is I think a good example of a leaky abstraction. Although it seems that the switch between signup layouts is based on a prop, we have penetrated that abstraction and included more step coupled styling logic. Furthermore these style injections have gone beyond the signup/step framework and has even been injected into stand alone components. 😬 

image

This results in high coupling and low [cohesion](https://en.wikipedia.org/wiki/Cohesion_(computer_science)) of code. A change to one aspect requires changes done to multiple several places. 

In other words signup components should not have any idea of what "steps, signup framework, stepper framework" are. At the very least, any overrides related to the steps needs to live closer to the steps. And since a prop (`isSocialFirst`) is introduced to control/switch the behaviour, these styles should activate based on this param not on step classes like `is-user-social`.

CC - @alshakero

**Comment by escapemanuele** [comment]
You are totally right @jdc91, that's solid and valid feedback. I'll keep it in mind!

**Review by aaronyan — COMMENTED** [review]
LGTM! I was able to assign myself and view experiences for `control` and `treatment` using the bookmarklet in Incognito + unblock third party cookies! Nice work @jdc91 at getting the experiences to load on the same URL.

**Review by southp — APPROVED** [review]
The code works as expected. How the variation works underneath is a bit hacky, but I think it's a balanced approach with all things considered. 🚢

**Review by southp — CHANGES_REQUESTED** [review]
Oh wait. Some pre-release E2E tests are failing. 
We will need to fix them before deploying.

File: packages/calypso-e2e/src/lib/components/domain-search-component.ts [hunk 1]
```diff
@@ -81,6 +81,8 @@ export class DomainSearchComponent {
 
 		await target.click();
 
+		await this.page.getByRole( 'button' ).filter( { hasText: 'Continue' } ).waitFor();
+
```
Comment (ddc22) [inline packages/calypso-e2e/src/lib/components/domain-search-component.ts:85]: @DustyReagan I am adding this line since there seems to be a delay the UI button that needs to be clicked.

**Review by southp — APPROVED** [review]
Leaving an explicit approval now that the e2e tests have been fixed.
